nativelib/src/main/resources/scala-native/gc/boehm/gc.c

```diff
@@ -61,6 +61,16 @@ size_t scalanative_GC_get_max_heapsize() {
     return Parse_Env_Or_Default("GC_MAXIMUM_HEAP_SIZE", heap_sz);
 }
 
+size_t scalanative_GC_get_used_heapsize() {
+    struct GC_prof_stats_s *stats =
+        (struct GC_prof_stats_s *)malloc(sizeof(struct GC_prof_stats_s));
+    GC_get_prof_stats(stats, sizeof(struct GC_prof_stats_s));
+    size_t heap_sz = stats->heapsize_full;
+    size_t unmapped_bytes = stats->unmapped_bytes;
+    free(stats);
+    return heap_sz - unmapped_bytes;
```

Comment:
Hm. Could we use `GC_get_heap_size()` instead?

Reply:
Probably it's thread-safe alternative might be better to use https://github.com/ivmai/bdwgc/blob/c3c44d85081044707eb8fa4f0db39e1fe6834853/include/gc/gc.h#L841-L850 
However, current solutions seems to be fine.